File: Backend/SkillForge/SkillForge/Models/DTOs/Tag/TagPageData.cs
using SkillForge.Models.DTOs.Article;
using SkillForge.Models.DTOs.User;

namespace SkillForge.Models.DTOs.Tag;

public class TagPageData
{
    public string Name { get; set; }

    public string? Description { get; set; }

    public int FollowersCount { get; set; }

    public int ArticlesCount { get; set; }

    public bool IsFollowedByCurrentUser { get; set; }

    public List<UserListItem> LatestFollowers { get; set; }

    public List<TopArticleItem> TopArticles { get; set; }

    public List<ArticleCard> LatestArticles { get; set; }
}

File: Frontend/skillforge/src/lib/types/TagPageData.ts
import type ArticleCardType from "./ArticleCardType";
import type TopArticleItemType from "./TopArticleItemType";
import type UserListItemType from "./UserListItemType";

export default interface TagPageData {
    Name: string,
    Description?: string,
    FollowersCount: number,
    ArticlesCount: number,
    IsFollowedByCurrentUser: boolean,
    LatestFollowers: UserListItemType[],
    LatestArticles: ArticleCardType[],
    TopArticles: TopArticleItemType[],
}